### PR TITLE
Refactored the S3 inactive to new approach

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

Refactored the S3 inactive to new approach

### CODE REVIEW

**Review by ebattat — COMMENTED** [review]
Pls fix conflict,
Have you tested it and verify that all working properly ?

**Comment by athiruma** [comment]
> Pls fix conflict, Have you tested it and verify that all working properly ?

yes, all are working.

File: cloud_governance/common/clouds/aws/s3/s3_operations.py [hunk 1]
```diff
@@ -325,3 +326,64 @@ def get_last_s3_policy_content(self, policy: str = '', file_name: str = '', s3_f
             os.system(f"gzip -d {local_file}")
             with open(os.path.join(temp_local_directory, file_name)) as f:
                 return f.read()
+
+    def list_buckets(self):
+        """
+        This method list all buckets
```
[the remark this answers is not in the document]
Comment (athiruma) [reply cloud_governance/common/clouds/aws/s3/s3_operations.py:332]: it is given by an AI assistant.

[the remark this answers is not in the document]
Comment (ebattat) [reply cloud_governance/common/clouds/aws/s3/s3_operations.py:332]: AI is wrong here

File: cloud_governance/common/clouds/aws/s3/s3_operations.py [hunk 2]
```diff
@@ -325,3 +326,64 @@ def get_last_s3_policy_content(self, policy: str = '', file_name: str = '', s3_f
...
+        :return:
+        """
+        try:
+            return self.__s3_client.list_buckets().get('Buckets', [])
+        except Exception as err:
+            return []
+
+    def get_bucket_tagging(self, bucket_name: str, **kwargs):
+        """
+        This method get tagging buckets
```
Comment (ebattat) [inline cloud_governance/common/clouds/aws/s3/s3_operations.py:342]: gets

File: cloud_governance/common/clouds/aws/s3/s3_operations.py [hunk 3]
```diff
@@ -325,3 +326,64 @@ def get_last_s3_policy_content(self, policy: str = '', file_name: str = '', s3_f
...
+        bucket_data = {}
+        try:
+            bucket_data = self.__s3_client.list_objects_v2(Bucket=bucket_name, Prefix=prefix, **kwargs)
+        except Exception as err:
+            logger.error(err)
+        return bucket_data
+
+    def get_bucket_contents(self, bucket_name: str, **kwargs):
+        """
+        This method get contents of a bucket
```
Comment (ebattat) [inline cloud_governance/common/clouds/aws/s3/s3_operations.py:370]: gets

File: cloud_governance/common/clouds/aws/s3/s3_operations.py [hunk 4]
```diff
@@ -325,3 +326,64 @@ def get_last_s3_policy_content(self, policy: str = '', file_name: str = '', s3_f
...
+        :param bucket_name:
+        :param kwargs:
+        :return:
+        """
+        bucket_data = self.list_objects_v2(bucket_name, **kwargs)
+        return bucket_data.get('Contents', [])
+
+    def get_bucket_location(self, bucket_name: str, **kwargs):
+        """
+        This method get location of a bucket
```
Comment (ebattat) [inline cloud_governance/common/clouds/aws/s3/s3_operations.py:380]: gets

**Review by ebattat — APPROVED** [review]

**Pull request merged**